web/src/engine/main/src/keymanEngine.ts

```diff
@@ -253,10 +255,27 @@ export default class KeymanEngine<
           this.core.resetContext(target);
         }
       },
-      predictionContext: new PredictionContext(this.core.languageProcessor, this.core.keyboardProcessor),
+      predictionContext: predictionContext,
       keyboardCache: this.keyboardRequisitioner.cache
     });
 
+    /*
+     * Handler for post-processing once a suggestion has been applied.
+     *
+     * This is called after the suggestion is applied but _before_ new
+     * predictions are requested based on the resulting context.
+     */
+    this.core.languageProcessor.on('suggestionapplied', () => {
```

Comment:
This new location really is a better placement for the listener, now that I've noticed it.

I always _did_ feel things were a bit messy with how I left `PredictionContext` during 17.0, but at least it _was_ enough to make the modularization splits needed to move forward at the time.  I'm almost certain that this change, along with what it enables, clears up a lot of my lingering concerns from then.